backend/audit/cog_agency.py

```diff
@@ -0,0 +1,93 @@
+# cog_over_assignment
+#   Input - federal_awards_data - federal awards json object
+#   Returns - cog_agency, over_agency - Agency prefix integer
+#
+# Algorithm:
+# Use all "federal_awards" to do calculations required to
+#   determine cognizant / oversight agency assignment
+
+# Calculate total amount_expended (tot_amount_expended)
+# Calculate total amount_expended by agency
+#   tot_amount_agency = {agency, tot_amt}
+# Calculate total direct award amount_expended by agency
+#   tot_da_amount_agency = {agency, tot_da_amt}
+# Calculate total direct award amount_expended (tot_da_amount_expended)
+#
+# If tot_amount_expended > $ 50,000,000, find cog agency
+#   Pull 2019 data from DB
+#       If 2019 data exists, use this for all calculations
+#           data_to_use = 2019 data
+#           Do all the above calculations with data_to_use
+#       else use current data
+#           data_to_use = current data
+#       if tot_da_amount_expended >= 25% tot_amount_expended
+#           cog_agency = agency with max tot_da_amt in tot_da_amount_agency
+#       else
+#           cog_agency = agency with max tot_amt in tot_amount_agency
+# else, find over agency
+#       if tot_da_amount_expended >= 25% tot_amount_expended
+#           over_agency = agency with max tot_da_amt in tot_da_amount_agency
+#       else
+#           over_agency = agency with max tot_amt in tot_amount_agency
+#############################################################
+
+
+def cog_over_assignment(federal_awards_data):
+    cog_agency = 0
+    over_agency = 0
+    tot_amount_expended = 0
+    tot_amount_agency = {}
+    tot_da_amount_agency = {}
+    tot_da_amount_expended = 0
+    for i in federal_awards_data["FederalAwards"]["federal_awards"]:
+        tot_amount_expended += i["program"]["amount_expended"]
+        agency = i["program"]["federal_agency_prefix"]
+        if agency in tot_amount_agency.keys():
+            tot_amount_agency[agency] += i["program"]["amount_expended"]
+        else:
+            tot_amount_agency[agency] = i["program"]["amount_expended"]
+        if i["direct_or_indirect_award"]["is_direct"] == "Y":
+            tot_da_amount_expended += i["program"]["amount_expended"]
+            if agency in tot_da_amount_agency.keys():
+                tot_da_amount_agency[agency] += i["program"]["amount_expended"]
+            else:
+                tot_da_amount_agency[agency] = i["program"]["amount_expended"]
+    #   print("tot_amount_expended = ", tot_amount_expended)
+
+    tot_amount_agency = list(
+        sorted(tot_amount_agency.items(), reverse=True, key=lambda item: item[1])
+    )
+    #   print("tot_amount_agency = ", tot_amount_agency)
+
+    tot_da_amount_agency = list(
+        sorted(tot_da_amount_agency.items(), reverse=True, key=lambda item: item[1])
+    )
+    #   print("tot_da_amount_agency = ", tot_da_amount_agency)
+
+    #   print("tot_da_amount_expended = ", tot_da_amount_expended)
+
+    if tot_amount_expended > 50000000:
```

Comment:
Lets remove all of the constants from this file, and put them out in `settings`. If these numbers ever change, I don't want to have to find them all over the codebase.

For now, it would be enough to have a set of constants at the top of the file in a dictionary. 

```
constants = {
   "amount_expended_limit" : 50000000,
   ...
}
```

or similar. We can pull them out of this file later.

I'd also like to suggest adding a few constants like

```
THOUSAND = 1000
HUNDRED_THOUSAND = 100000
MILLION = 1000000
```

so that we can write

```
constants = {
   "amount_expended_limit" : 50 * MILLION,
   ...
}
```

which makes it much easier to glance and see what the numeric constant is.